### DISCUSSION TITLE
Spacy project assets doesn't update file

### DISCUSSION THREAD

**Discussion by Shiyinq** [explosion/spaCy#12180] [Help: Coding & Implementations] [answered]
i'm using spacy project and for the file i put on my storage and pull with command
`spacy project assets`
output:
 ✔ Downloaded asset 
 ✔ Downloaded asset 

for the first time it work, it will download 2 file and put on folder assets

after that i update the data to file in my storage lets say i add 1000 row data

and then i want to update my file in spacy project, so i run the command again
`spacy project assets`
output:
 ✔ Downloaded asset 
 ✔ Downloaded asset 

nothing error after run command, but my file in folder assets not updated, why?
how to update or replace everytime  fetch the project assets?

**Comment by adrianeboyd** [comment, marked as answer]
It currently doesn't download the file again if it exists and is not empty (since sometimes failed downloads lead to empty files).

You can add/update a `checksum` for the asset to have a way to indicate that the asset has changed and should be downloaded again.

It might make sense to add a `--force` option here as with `spacy project run` to redownload all assets.
